cylc/rose/platform_utils.py

```diff
@@ -76,6 +83,17 @@ def get_platform_from_task_def(flow: str, task: str) -> Dict[str, Any]:
     return platform
 
 
+def get_compat_mode(run_dir: Union[str, Path]) -> bool:
+    """Check whether this is a Cylc 7 Back compatibility mode workflow:
+
+    See https://github.com/cylc/cylc-rose/issues/319
+
+    Args:
+        run_dir: Cylc workflow run directory.
+    """
+    return (Path(run_dir) / WorkflowFiles.SUITE_RC).exists()
```

Comment:
Couldn't we just import and use `cylc.flow.workflow_files.check_deprecation`?

Reply:
I guess that we could, but I'd rather not risk change the state of the `cylc.flow.flags.cylc7_back_compat` flag if at all possible: This approach means that information is only really flowing from Cylc flow into Cylc Rose.